### PR TITLE
examples: Add UDP proxy sandbox example

### LINKED ISSUE #15188: Add UDP proxy sandbox example
*Description*:

It would be great if we had a sandbox example of proxying UDP

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: examples: Add UDP proxy sandbox example
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Fix #15188
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/master/api/review_checklist.md):]

### CODE REVIEW

**Comment by alyssawilk** [comment]
Nice addition!  Please ping when you are ready for review :-)

**Comment by phlax** [comment]
will do - the config is not valid to the protobuf spec i think - and the pr was partly to demonstrate that - ill update shortly

**Comment by phlax** [comment]
@mattklein123 @alyssawilk just waiting for ci to render the docs but i think this should be good for review

**Comment by phlax** [comment]
docs are here https://storage.googleapis.com/envoy-pr/b4f62cf/docs/start/sandboxes/udp.html

File: docs/root/start/sandboxes/udp.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+   $ docker-compose logs service-udp
+   Attaching to udp_service-udp_1
+   service-udp_1  | Listening on UDP port 5005
+   service-udp_1  | HELO
+   service-udp_1  | OLEH
+
+Step 4: View the Envoy admin ``UDP`` stats
+******************************************
+
+You can view the ``UDP``-related admin stats provided by the Envoy admin endpoint.
```
Comment (phlax) [inline docs/root/start/sandboxes/udp.rst:70]: ...related stats  provided by...

**Review by mattklein123 — APPROVED** [review]
Awesome!

**Pull request merged**